vmhost/contexts/runtime.go

```diff
@@ -480,6 +480,7 @@ func (context *runtimeContext) SetVMInput(vmInput *vmcommon.ContractCallInput) {
 	if len(vmInput.CallerAddr) > 0 {
 		context.vmInput.CallerAddr = make([]byte, len(vmInput.CallerAddr))
 		copy(context.vmInput.CallerAddr, vmInput.CallerAddr)
+		context.vmInput.OriginalCallerAddr = context.vmInput.CallerAddr
```

Comment:
OriginalCaller should be here as well. and do like a deep copy. - outside of the if.

Reply:
fixed